spatialmath/box_test.go

```diff
@@ -51,3 +51,16 @@ func TestBoxVertices(t *testing.T) {
 	test.That(t, R3VectorAlmostEqual(vertices[6], r3.Vector{-1, -1, 1}.Add(offset), 1e-8), test.ShouldBeTrue)
 	test.That(t, R3VectorAlmostEqual(vertices[7], r3.Vector{-1, -1, -1}.Add(offset), 1e-8), test.ShouldBeTrue)
 }
+
+func TestBoxPC(t *testing.T) {
+	offset := r3.Vector{0, 0, 0}
+	dims := r3.Vector{2, 2, 2}
+	orien := &Quaternion{1, 1, 0, 0}
+	pose := NewPoseFromOrientation(offset, orien)
+	box := &box{pose, [3]float64{0.5 * dims.X, 0.5 * dims.Y, 0.5 * dims.Z}, 10, ""}
+	myMap := make(map[string]interface{})
+	myMap["xIter"] = 0.5
+	myMap["yIter"] = 0.5
+	myMap["zIter"] = 0.5
+	box.ToPointCloud(myMap)
```

Comment:
We'll need to think about how to test this better, because as is this will not catch any changes that would break your function

Reply:
based off the dimensions we can calculate how many points the r3.vector list contains. because we also have the center and rotation matrix we can apply that to the existing vertices and check within the r3.vector list for each corresponding vertex

Reply:
What I would like to see as a test for this, would look like, say, a 2x2x2 box, a 1x1.5x4 box, and a radius=2.5 sphere, at hardcoded poses, with the expected set of points also hardcoded.